library/n2_protocol.hpp

```diff
@@ -29,9 +29,22 @@ struct n2_request {
 struct n2_repliers {
 	// Here refcounting isn't required, but shared_ptr is used for dynamic destruction and for compatibility
 	// with binding to std::function (bound parameter must be copyable).
-	std::function<int (const std::shared_ptr<n2_body> &)> on_reply;
+	std::function<int (const std::shared_ptr<n2_body> &/*body*/, bool /*last*/)> on_reply;
 
-	std::function<int (int)> on_reply_error;
+	std::function<int (int /*err*/)> on_reply_error;
+
+	// Helpers
+	inline int on_last_reply(const std::shared_ptr<n2_body> &body) const {
```

Comment:
(only notice) Since user now calls only these inline methods, we could make `std::function`s to be private. Then, evolving, we could directly derive `xx_repliers` (e.g. `lookup_repliers`) from `n2_repliers`, and so we fully replace `std::function`s by virtual methods. Now, you see, `std::function`s become superfluous proxy entities. We could think about it later.

Reply:
Now we use `n2_repliers::on_reply` too. Ok, will think about it.